content/en/docs/topics/kubernetes_apis.md

```diff
@@ -0,0 +1,127 @@
+---
+title: "Kubernetes APIs"
+description: "Explains Kubernetes APIs in Helm templates"
+aliases: ["docs/k8s_apis/"]
+---
+
+Kubernetes is an API-driven system and the API evolves over time to reflect
+the evolving understanding of the problem space. This is common practice
+across systems and their APIs. An important part of evolving APIs is a good
+deprecation policy and process to inform users of how changes to APIs are
+implemented. In other words, consumers of your API need to know in advance and
+in what release an API will be removed or changed. This removes the element of
+surprise and breaking changes to consumers.
+
+The [Kubernetes deprecation policy](https://kubernetes.io/docs/reference/using-api/deprecation-policy/)
+documents how Kubernetes handles the changes to its API versions. The policy
+for deprecation states the timeframe that API versions will be supported following
+a deprecation announcement. It is therefore important to be aware of deprecation
+announcements and know when API versions will be removed, to help mimimize the
+effect.
+
+This is an example of an announcement [for the removal of deprecated API versions in Kubernetes 1.16](https://kubernetes.io/blog/2019/07/18/api-deprecations-in-1-16/)
+and was advertised a few months prior to the release. These API versions would
+have been announced for deprecation prior to this again. This shows that there
+is a good policy in place which informs consumers of API version support. 
+
+Helm templates specify a [Kubernetes API group](https://kubernetes.io/docs/concepts/overview/kubernetes-api/#api-groups)
+when defining a Kubernetes object, similar to a Kubernetes manifest file. It is
+specified in the `apiVersion` field of the template and it identifies the API
+version of the Kubernetes object. This means that Helm users and chart maintainers
+need to be aware when Kubernetes API versions have been deprecated and in what
+Kubernetes version they will be removed.
+
+## Chart Maintainers
+
+You should audit your charts checking for Kubernetes API versions that are
+deprecated or are removed in a Kubernetes version. The API versions found out of
+support should be updated to the supported version and a new version of the chart
+released. The API version is defined by the `kind` and `apiVersion` fields. For
+example, here is a removed `Deployment` object API version in Kubernetes 1.16:
+
+```yaml
+apiVersion: apps/v1beta1
+kind: Deployment
+```
+
+## Helm Users
+
+You should audit the charts that you use (similar to the [chart mantainer](#chart-maintainers))
+and identify any charts where API verions are deprecated or removed in a
+Kubernetes version. For the charts identified, you need to check for the latest
+version of the chart (which has supported API versions) or update the chart yourself.
```

Comment:
I agree with you in principle. However, I feel it is a better practice to update charts where possible when API versions are in deprecation mode than when the API versions are removed. It would help minimise the issue that are occurring with Helm. I am trying to demonstrate best practices here, not dictate to users.